File: Backend/database/app/utils/mysql_health.py
"""
Shared MySQL health-score formula — backend mirror of
frontend/src/utils/mysqlHealth.js. Used by mysql_report_service.py so the
PDF/scheduled-email report shows the exact same score the Dashboard/Reports
page renders, per the spec's "use the same health thresholds... do not
create different health logic" requirement.
"""


def compute_health_score(long_running_count: int, conn_pct: float, cache_pct: float) -> int:
    score = 100
    if conn_pct > 90:
        score -= 30
    elif conn_pct > 70:
        score -= 15
    if cache_pct < 80:
        score -= 20
    elif cache_pct < 90:
        score -= 10
    if long_running_count > 5:
        score -= 15
    elif long_running_count > 0:
        score -= 5
    return max(0, score)

File: frontend/src/utils/mysqlHealth.js
/**
 * Shared MySQL health-score formula — used by both the Dashboard and the
 * Reports page so the two never diverge (previously each computed its own,
 * different score). `longRunningCount` is the ACTIVE long-running-query
 * count; historical report periods that don't have this data pass 0, which
 * simply drops that term's penalty rather than fabricating a count.
 */
export function computeHealthScore(longRunningCount, connPct, cachePct) {
  let score = 100;
  if (connPct > 90) score -= 30;
  else if (connPct > 70) score -= 15;
  if (cachePct < 80) score -= 20;
  else if (cachePct < 90) score -= 10;
  if (longRunningCount > 5) score -= 15;
  else if (longRunningCount > 0) score -= 5;
  return Math.max(0, score);
}
